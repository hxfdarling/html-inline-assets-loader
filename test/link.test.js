
import 'jest-extended';

const { compile, evaluated } = require('./helpers');

describe('link loader', () => {
  it('should link js/css', async () => {
    const stats = await compile('link.html');

    expect(stats.compilation.warnings).toBeArrayOfSize(0);
    expect(stats.compilation.errors).toBeArrayOfSize(0);

    const { modules } = stats.toJson();
    const [module] = modules;
    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules))
      .toMatchSnapshot('module (evaluated)');
  });
});
