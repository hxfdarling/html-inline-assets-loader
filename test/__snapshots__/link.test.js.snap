// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link loader should link js/css: module (evaluated) 1`] = `
"<!DOCTYPE html><html><head>
    <title>imt application</title>
    <link rel=\\"stylesheet\\" href=\\"inline_02f276.css\\">
    <script src=\\"inline_fde1ab.js\\"></script>
  </head>
  <body>

</body></html>"
`;

exports[`link loader should link js/css: module 1`] = `"module.exports = \\"<!DOCTYPE html><html><head>\\\\n    <title>imt application</title>\\\\n    <link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"inline_02f276.css\\\\\\">\\\\n    <script src=\\\\\\"inline_fde1ab.js\\\\\\"></script>\\\\n  </head>\\\\n  <body>\\\\n\\\\n</body></html>\\";"`;
